distributed/comm/tests/test_comms.py

```diff
@@ -201,6 +203,26 @@ def test_get_local_address_for():
 #
 
 
+@pytest.mark.asyncio
+async def test_tcp_listener_does_not_call_handler_on_handshake_error():
+    handle_comm_called = False
+
+    async def handle_comm(comm):
+        nonlocal handle_comm_called
+        handle_comm_called = True
+
+    with dask.config.set({"distributed.comm.timeouts.connect": 0.01}):
+        listener = await tcp.TCPListener("127.0.0.1", handle_comm)
+        host, port = listener.get_host_port()
+        # connect without handshake:
+        reader, writer = await asyncio.open_connection(host=host, port=port)
+        await asyncio.sleep(0.02)
```

Comment:
I'm curious what prompted the addition of this `asyncio.sleep`

Reply:
This ensures we actually hit the timeout of the handshake path on the server side. Without this, the test would be useless. I'll add a comment.